### ISSUE TITLE
no-useless-await: false negative

### ISSUE THREAD

**Issue by creage** [playwright-community/eslint-plugin-playwright#147]
I have a custom component with static async method:

```ts
export default class Custom {
    static async isClosed(selector: string) {
        return page.waitForSelector(selector, { state: 'detached'});
    }
}
```

Any call to that method fires me an `Unnecessary await expression. This method does not return a Promise`

```ts
await Custom.isClosed('.my-selector');
```

Apparently because the name of the method `isClosed` conflicts with the same method of Playwright.
